lightning/src/onion_message/functional_tests.rs

```diff
@@ -546,6 +547,30 @@ fn too_big_packet_error() {
 	assert_eq!(err, SendError::TooBigPacket);
 }
 
+#[test]
+fn blinded_path_padding() {
+	// Make sure that for a blinded path, all encrypted payloads are padded to equal lengths.
+	let nodes = create_nodes(4);
+	let test_msg = TestCustomMessage::Pong;
+
+	let secp_ctx = Secp256k1::new();
+	let intermediate_nodes = [
+		MessageForwardNode { node_id: nodes[1].node_id, short_channel_id: None },
+		MessageForwardNode { node_id: nodes[2].node_id, short_channel_id: None },
```

Comment:
Interestingly, if we use the compact representation by giving `Some(42)` for `short_channel_id`, then all hops will have the same size without padding. Let's update the context below to use `vec![0u8; 42]` such that the receive TLVs are longer even in that situation.

Reply:
Updated the context in [pr3177.17](https://github.com/shaavan/rust-lightning/commits/pr3177.17) to ensure the size of hops vary before padding!
Thanks for the pointer!